oxenc/bt_producer.h

```diff
@@ -299,19 +334,38 @@ class bt_list_producer {
         return bs->init + next + 1;
     }
 
+    template <string_like T>
+    void append(const T& data) {
+        if (has_child)
+            throw std::logic_error{"Cannot append to list when a sublist is active"};
+        append_impl(data.data(), data.size());
+        append_intermediate_ends();
+    }
```

Comment:
This overload seems unnecessary if the immediately following template becomes `const_span_convertible` instead of `char_span_convertible`